### ISSUE TITLE
Probable mesh duplication warning

### ISSUE THREAD

**Issue by rjshae** [xoreos/xoreos#412]
I'm seeing the following warnings when I start up NWN2. These appear to be a few of the models used in the starting area, although RRRR is the roof tile and the other tile types in use are missing. (The ellipses are mine.)

`WARNING: Warning: probable mesh duplication of: tl_SI_RRRR_01.xoreos.default.TL_SI_RRRR_01, attempting to correct!
...
WARNING: Warning: probable mesh duplication of: tl_SI_RRRR_01.xoreos.default.TL_SI_RRRR_01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CHAIR01.xoreos.default.PLC_MC_CHAIR01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_IPoint.xoreos.default.PLC_IPoint, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_IPoint.xoreos.default.PLC_IPoint, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_IPoint.xoreos.default.PLC_IPoint, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_SHELF01.xoreos.default.PLC_MC_SHELF01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_SHELF01.xoreos.default.PLC_MC_SHELF01A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CABINET1.xoreos.default.PLC_MC_CABINET1, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CHAIR01.xoreos.default.PLC_MC_CHAIR01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CHAIR01.xoreos.default.PLC_MC_CHAIR01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CHAIR01.xoreos.default.PLC_MC_CHAIR01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CHAIR01.xoreos.default.PLC_MC_CHAIR01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_SHELF03.xoreos.default.PLC_MC_SHELF03, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_SHELF03.xoreos.default.PLC_MC_SHELF03A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CANBRA01.xoreos.default.PLC_MC_CANBRA01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CANBRA01.xoreos.default.PLC_MC_CANBRA01A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BASKET01.xoreos.default.PLC_MC_Basket01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BASKET01.xoreos.default.PLC_MC_Basket01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CRATE09.xoreos.default.PLC_MC_CRATE09, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BARREL01.xoreos.default.PLC_MC_BARREL01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BARREL01.xoreos.default.PLC_MC_BARREL01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BARREL01.xoreos.default.PLC_MC_BARREL01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BARREL01.xoreos.default.PLC_MC_BARREL01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BARREL01.xoreos.default.PLC_MC_BARREL01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_Basket05.xoreos.default.PLC_MC_Basket05, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BASKET01.xoreos.default.PLC_MC_Basket01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CHEST03.xoreos.default.PLC_MC_CHEST03, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP01.xoreos.default.PLC_MC_LAMP01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_CRATE09.xoreos.default.PLC_MC_CRATE09, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_HPLANT04.xoreos.default.PLC_MC_HPLANT04, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_BARREL01.xoreos.default.PLC_MC_BARREL01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_HPLANT01.xoreos.default.PLC_MC_HPLANT01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_SHELF05.xoreos.default.PLC_MC_SHELF05, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_RUG01.xoreos.default.PLC_MC_RUG01, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_HPLANT04.xoreos.default.PLC_MC_HPLANT04, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_MC_LAMP03.xoreos.default.PLC_MC_LAMP03A, attempting to correct!
WARNING: Warning: probable mesh duplication of: PLC_DT_StdInt01.xoreos.default.PLC_DT_STDINT01, attempting to correct!
WARNING: Warning: probable mesh duplication of: P_EEM_NK_GLOVES01.xoreos.default.P_EEM_NK_GLoves01, attempting to correct!`

**Comment by DrMcCoy** [comment]
Yeah, these are debug warnings @mirv-sillyfish introduced for the new renderer.

@mirv-sillyfish, maybe we can change the `warning()`s into `debugC()`s, so that they're only printed when debug information is enabled on the command line?

**Comment by rjshae** [comment]
I noticed the recent merge of "Force add duplicate mesh names". However, I believe NWN2 re-uses existing meshes, such as for tiles of the same type or speed trees with the same seed. I suspect we could end up with a lot more memory usage if renamed duplicate meshes are allowed as a default.

**Comment by mirv-sillyfish** [comment]
yes, probably should turn it into debug message instead.
The idea is that individual "games" (or subclasses of modelnode) should check for themselves if they should force duplicate, or de-duplicate. Unfortunately, some have the same names but are entirely different meshes. I've not yet figured out a universal way of knowing that - there likely is one, but it's not as obvious as it could be.

So default to allow duplicate meshes to avoid the memory leaks, and it can then be dealt with individually later on.
